### PR TITLE
add predict_step to BYOLTask

### PR DESCRIPTION
**Pull request by isaaccorley**
Related to #813. This PR adds a `predict_step` method to `BYOLTask` so that users can utilize PyTorch Lightning's `trainer.predict()` function which will automatically loop and predict over a PyTorch DataLoader e.g.:

```python
preds = trainer.predict(model=task, dataloaders=datamodule.test_dataloader())

# Or if your datamodule has a `predict_dataloader` method defined
preds = trainer.predict(model=task, datamodule=datamodule)
```

Note the way the EncoderWrapper is setup is that calling `self(x)` on a batch of images will return the projector output embeddings. However, we only need this during training. For linear evaluation or other downstream tasks we want the output of the image encoder (the output of the avg pool layer of the ResNet). To work around this I had to add another hook to store the embeddings on a forward pass in the `self._embeddings` attribute.

### CODE REVIEW

File: torchgeo/trainers/byol.py [hunk 1]
```diff
@@ -449,3 +452,16 @@ def validation_step(self, *args: Any, **kwargs: Any) -> None:
...
+        """Compute and return the output embeddings of the image encoder.
+
+        Args:
+            batch: the output of your DataLoader
+        Returns:
+            image embeddings
+        """
+        batch = args[0]
+        x = batch["image"]
+        _ = self(x)
```
Comment (adamjstewart) [inline torchgeo/trainers/byol.py:466]:
```suggestion
        self(x)
```

File: torchgeo/trainers/byol.py [hunk 2]
```diff
@@ -449,3 +452,16 @@ def validation_step(self, *args: Any, **kwargs: Any) -> None:
 
     def test_step(self, *args: Any, **kwargs: Any) -> Any:
         """No-op, does nothing."""
+
+    def predict_step(self, *args: Any, **kwargs: Any) -> Tensor:
+        """Compute and return the output embeddings of the image encoder.
+
+        Args:
+            batch: the output of your DataLoader
```
Comment (adamjstewart) [inline torchgeo/trainers/byol.py:460]: I would add a newline here

File: torchgeo/trainers/byol.py (hunk 1 above)
Comment (isaaccorley) [reply torchgeo/trainers/byol.py:466]: I believe this would print the outputs which is not what we want.

Comment (isaaccorley) [reply torchgeo/trainers/byol.py:466]: I was wrong, it doesn't seem to print the outputs.

Comment (adamjstewart) [reply torchgeo/trainers/byol.py:466]: It isn't a Jupyter Notebook...

Comment (isaaccorley) [reply torchgeo/trainers/byol.py]: Fixed

File: torchgeo/trainers/byol.py (hunk 2 above)
Comment (isaaccorley) [reply torchgeo/trainers/byol.py:460]: Fixed

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**